serialization-protobuf/src/main/kotlin/io/github/kkafka/consumer/flowProtobuf.kt

```diff
@@ -0,0 +1,21 @@
+package io.github.kkafka.consumer
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.protobuf.ProtoBuf
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import java.time.Duration
+
+public fun <K, V> KafkaConsumer<ByteArray, ByteArray>.pollWithFlow(
+    pollingPeriod: Duration,
+    keyDeserializer: DeserializationStrategy<K>,
+    valueDeserializer: DeserializationStrategy<V>,
+    protobuf: ProtoBuf = ProtoBuf
+): Flow<ConsumerRecord<K, V>> = pollWithFlow(pollingPeriod)
+    .map {
+        val key = protobuf.decodeFromByteArray(keyDeserializer, it.key())
+        val value = protobuf.decodeFromByteArray(valueDeserializer, it.value())
+        ConsumerRecord(it.topic(), it.partition(), it.offset(), key, value)
+    }
```

Comment:
Do we also want to add a protobuf override of `pollWithFlowByBatches`?

Reply:
not really because I have no way to reconstruct a `ConsumerRecords` class produced by kafka. This is because it comes with extra info (like a checksum) and I am unsure of what behaviour a user would expect there